### PR TITLE
fix bugs for importing and exporting samples

### PR DESCRIPTION
**Pull request by adambasha0**
- [x] rather 1-story 1-commit than sub-atomic commits

- [x] commit title is meaningful =>  git history search

- [x] commit description is helpful => helps the reviewer to understand the changes

- [x] code is up-to-date with the latest developments of the target branch (rebased to it or whatever) => :fast_forward:-merge for linear history is favoured

- [x] added code is linted

- [ ] tests are passing (at least locally): we still have some random test failure on CI. thinking of asking spec/examples.txt to be commited

- [ ] in case the changes are visible to the end-user,  video or screenshots should be added to the PR => helps  with user testing

- [ ] testing coverage improvement is improved.

- [ ] CHANGELOG :  add a bullet point on top (optional: reference to github issue/PR )

- [ ] parallele PR for documentation  on docusaurus  if the feature/fix is tagged for a release

### CODE REVIEW

**Review by FabianMauz — COMMENTED** [review]
only first part, try to add some tests to get more insight into the logic

File: lib/export/export_excel.rb [hunk 1]
```diff
@@ -119,9 +119,14 @@ def filter_with_permission_and_detail_level(sample)
       # return all data if sample in own collection
       if sample['shared_sync'] == 'f' || sample['shared_sync'] == false
         headers = @headers
+        comparison_values = ['melting pt', 'boiling pt']
```
Comment (FabianMauz) [inline lib/export/export_excel.rb:122]: the name comparison_values is for me a little bit misleading. Maybe interval_values or intervaluable_values is clearer.

File: lib/import/import_samples.rb [hunk 2]
```diff
@@ -177,6 +177,15 @@ def molecule_not_exist(molecule)
     molecule.nil?
   end
 
+  # format row[field] for melting and boiling point
+  def process_row(row_field)
```
Comment (FabianMauz) [inline lib/import/import_samples.rb:181]: i would rename method by a more meaningful name, maybe format_to_interval_syntax or something like that

File: lib/import/import_samples.rb [hunk 3]
```diff
@@ -177,6 +177,15 @@ def molecule_not_exist(molecule)
...
+    # Regex checks for a range of numbers that are separated by a dash, or a single number
+    matches = row_field.scan(/^(-?\d+(?:[.,]\d+)?)(?:\s*-\s*(-?\d+(?:[.,]\d+)?))?$/).flatten.compact
```
Comment (FabianMauz) [inline lib/import/import_samples.rb:183]: the regex is quite complicated to read but i have no idea how to make it more readable :)

**Review by PiTrem — APPROVED** [review]

**Pull request merged**